Sources/Testing/Support/CustomIssueRepresentable.swift

```diff
@@ -0,0 +1,86 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024–2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+/// A protocol that provides instances of conforming types with the ability to
+/// record themselves as test issues.
+///
+/// When a type conforms to this protocol, values of that type can be passed to
+/// ``Issue/record(_:_:)``. The testing library then calls the
+/// ``represent(as:)`` function and passes it an instance of ``Issue`` that will
+/// be used to represent the value. The function can then reconfigure or replace
+/// the issue as needed.
+///
+/// This protocol may become part of the testing library's public interface in
+/// the future. There's not really anything _requiring_ it to conform to `Error`
+/// but all our current use cases are error types. If we want to allow other
+/// types to be represented as issues, we will need to add an overload of
+/// `Issue.record()` that takes `some CustomIssueRepresentable`.
+protocol CustomIssueRepresentable: Error {
+  /// Customize the issue that will represent this value.
+  ///
+  /// - Parameters:
+  ///   - issue: The issue to customize. The function consumes this value.
+  ///
+  /// - Returns: A customized copy of `issue`.
+  func customize(_ issue: consuming Issue) -> Issue
```

Comment:
It could be useful to return Optional, so that an error could elect to suppress recording an issue representing itself

Reply:
That would require us to change `Issue.record()` to return an optional value, which would be API- and ABI-breaking. If we made this a public feature, we might want to do that, but it's not helpful for the current use case. Make sense?

Reply:
Sounds good. If we do move forward with promoting this to public, I do think we ought to consider it

Reply:
Fair enough.